client/my-sites/comments/comment-list/index.jsx

```diff
@@ -176,14 +151,13 @@ export class CommentList extends Component {
 				>
 					{ map( filteredComments, comment =>
 						<CommentDetail
-							commentId={ comment.ID }
-							deleteForever={ this.deleteForever( comment.ID ) }
+							comment={ comment }
+							deleteCommentPermanently={ this.deleteCommentPermanently( comment.ID ) }
```

Comment:
@gwwar was there a typo in that comment? it looks the same basically as what I see written in the PR.

that being said…

> avoid the function creation here

please! it should stand out that `deleteCommentPermanently` does not look like the others (`() => () => {}` instead of `() => {}`). It's okay to say, "this function I'm giving you takes a comment ID" and let the `<CommentDetail />` call it given its passed `comment`'s ID. at the end of the line we have custom data attributes to store identifier values for the actual markup receiving the click.